storage/engine/mvcc.go

```diff
@@ -41,38 +40,73 @@ const (
 var (
 	// MVCCKeyMax is a maximum mvcc-encoded key value which sorts after
 	// all other keys.
-	MVCCKeyMax = MVCCEncodeKey(roachpb.KeyMax)
+	MVCCKeyMax = MakeMVCCKey(roachpb.KeyMax)
+	// NilKey is the nil MVCCKey.
+	NilKey = MVCCKey{}
 )
 
 // MVCCKey is an encoded key, distinguished from roachpb.Key in that it
 // is an encoded version.
-type MVCCKey []byte
+type MVCCKey struct {
+	Key       roachpb.Key
+	Timestamp roachpb.Timestamp
+}
+
+// MakeMVCCKey creates an MVCCKey from a roachpb.Key.
+func MakeMVCCKey(key roachpb.Key) MVCCKey {
+	return MVCCKey{Key: key}
+}
 
-// Next returns the next key in lexicographic sort order.
+// Next returns the next key.
 func (k MVCCKey) Next() MVCCKey {
-	return MVCCKey(roachpb.BytesNext(k))
+	ts := k.Timestamp.Prev()
```

Comment:
Good point about the panic, though that doesn't actually trigger in any of the current tests and won't ever happen in real life. I'll update the docs to clarify which operations are ok. 

We do see `k.Timestamp.Prev()` return 0 in tests. I can't recall which one now, though I suppose if I disable this it would show up.